site/components/pages/Resources/ResourcesList/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FC, useState } from "react";
+import { Trans, t } from "@lingui/macro";
+import { useForm, SubmitHandler } from "react-hook-form";
+import SearchIcon from "@mui/icons-material/Search";
+import { Text, Section, ButtonPrimary } from "@klimadao/lib/components";
+
+import { Card } from "components/Card";
+import { PodcastCard } from "components/PodcastCard";
+import { InputField } from "components/Form";
+import { getResourcesListErrorMap } from "../lib/getResourcesListErrorTranslations";
+
+import { fetchCMSContent } from "lib/fetchCMSContent";
+
+import * as styles from "./styles";
+import { Document } from "lib/queries";
+
+export interface Props {
+  documents: Document[];
+}
+
+type FormValues = {
+  search: string;
+};
+
+export const ResourcesList: FC<Props> = ({ documents }) => {
+  const [visibleDocuments, setVisibleDocuments] =
+    useState<Document[]>(documents);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [noResult, setNoResult] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>();
+
+  const onSubmit: SubmitHandler<FormValues> = async (values: FormValues) => {
+    setSearchTerm(values.search);
+
+    // reset list to default documents on empty string
+    if (!values.search) {
+      setVisibleDocuments(documents);
+      return;
+    }
+
+    try {
+      const searchResult = await fetchCMSContent("searchByText", {
+        searchQuery: values.search,
+      });
+
+      if (searchResult.length) {
+        setVisibleDocuments(searchResult);
+      } else {
+        setNoResult(true);
```

Comment:
When it shows the "no result found" message, the `props.documents` should not be rendered (currently, they are)

![image](https://user-images.githubusercontent.com/88635679/187982771-b679e64d-c408-4d8d-81d8-dfd412580500.png)

Related to this, maybe we can just set `visibleDocuments` to `[]` or null, and then we don't have to manage a second useState for `noResult`?

Reply:
Done

Reply:
In general we should discuss again what to expect on using all form elements.
Please see this comment in the ticket => https://github.com/KlimaDAO/klimadao/issues/534